### PR TITLE
Fix missing ttype validation

### LINKED ISSUE #3090: Bad error handling in /ttype/<type> endpoint
We observed a bad error handling when requesting the `/ttype/<type>` endpoint.

In specific, we faced the following error:
```
[ERROR][privacyidea.app:1892] Exception on /ttype/push"}. [GET]
...
AttributeError: 'NoneType' object has no attribute 'api_endpoint'
```
Actually, we could fix the problem but it would be nice to fix this upstream right away.

### Top-level intent

Access the `/ttype/<type>` endpoint.

### Steps to reproduce

1. Query `/ttype/test` endpoint
2. There will be a NoneType error in the logs.

### Expected outcome

Proper error handling

### Actual outcome

NoneType exception.

### Configuration

* **privacyIDEA version**: v3.6.3
* **Installation method**: (from Ubuntu packages, github, PyPI, ...)
* **Python version**: 3
* **Operating system**: linux
* **Webserver**: apache

### Log file

**Set PI_LOGLEVEL = logging.DEBUG in pi.cfg and take a look at the privacyidea.log!**
**If appropriate, attach the log file or paste relevant portions.**

### PR DESCRIPTION
**Pull request by Happy2C0de**
Fixes #3090

### CODE REVIEW

File: privacyidea/lib/config.py [hunk 1]
```diff
@@ -43,6 +43,7 @@
 from .log import log_with
 from ..models import (Config, db, Resolver, Realm, PRIVACYIDEA_TIMESTAMP,
                       save_config_timestamp, Policy, EventHandler)
+from privacyidea.lib.error import ValidateError
```
Comment (cornelinux) [inline privacyidea/lib/config.py:46]: I think we should use a `ParameterError`. The `ValidateError` is used, if the some error occurred during the authentication process.

In a normal authentication process you the application would not get the idea to ask for a non-existing tokentype, so this is rather an application passing wrong parameters.

File: privacyidea/lib/config.py [hunk 2]
```diff
@@ -451,6 +452,9 @@ def get_token_class(tokentype):
             tokenclass = tclass
             break
 
+    if tokenclass is None:
+        raise ValidateError("invalid tokentype provided. ttype: {}".format(tokentype.lower()))
```
Comment (cornelinux) [inline privacyidea/lib/config.py:456]: If I would call 

    /ttype/ööäääh

raising the ValidateError would raise an EncodingError. So you should do

~~~~python
raise ParameterError(u"Invalid tokentype provided. ttype: {}".format(tokentype.lower()))
~~~~

**Review by cornelinux — CHANGES_REQUESTED** [review]
Please issue your PR against the branch-3.7, this way the fix will be published with the next release 3.7.1.

File: privacyidea/lib/config.py (hunk 1 above)
Comment (Happy2C0de) [reply privacyidea/lib/config.py]: Done, good input. I also thought about ParameterError but didn't find an explanation.
Changed.

File: privacyidea/lib/config.py (hunk 2 above)
Comment (Happy2C0de) [reply privacyidea/lib/config.py]: Adjusted

**Comment by Happy2C0de** [comment]
@cornelinux 
I adjusted the error type and rebased the branch to `branch-3.7` in order to adjust the target branch.
LGTM

**Review by cornelinux — APPROVED** [review]

File: privacyidea/lib/config.py [hunk 3]
```diff
@@ -451,6 +452,9 @@ def get_token_class(tokentype):
             tokenclass = tclass
             break
 
+    if tokenclass is None:
+        raise ParameterError(u"Invalid tokentype provided. ttype: {}".format(tokentype.lower()))
```
Comment (cornelinux) [inline privacyidea/lib/config.py:456]: @Happy2C0de  It looks as if your change is too low level here, that causes several tests to fail!

Comment (Happy2C0de) [reply privacyidea/lib/config.py]: @cornelinux 
I moved the error handling to `ttype.py`.
LGTM

**Pull request merged**